reader.go

```diff
@@ -176,6 +176,59 @@ func (r *Reader) ReadMessage(ctx context.Context) (Message, error) {
 	}
 }
 
+// ReadLag returns the current lag of the reader by fetching the last offset of
+// the topic and partition and computing the difference between that value and
+// the offset of the last message returned by ReadMessage.
+//
+// This method is intended to be used in cases where a program may be unable to
+// call ReadMessage to update the value returned by Lag, but still needs to get
```

Comment:
For example, when the consumer is not ready to process the next message, correct?